app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivityViewModel.kt

```diff
@@ -113,18 +114,25 @@ class MainActivityViewModel(
     fun onNewIntent(intent: Intent?) {
         when {
             intent.hasPendingAuthorizationData -> {
+                val authorizationIdentifier = AuthorizationIdentifier(
+                    authorizationID = intent.authorizationId,
+                    connectionID = intent.connectionId
+                )
                 onShowAuthorizationDetailsEvent.postValue(
-                    ViewModelEvent(AuthorizationIdentifier(
-                        authorizationID = intent.authorizationId,
-                        connectionID = intent.connectionId
-                    ))
+                    ViewModelEvent(Bundle().apply {
+                        putSerializable(KEY_ID, authorizationIdentifier)
+                    })
                 )
             }
             intent.hasDeepLinkData -> {
                 intent.deepLink.extractConnectAppLinkData()?.let { connectionAppLinkData ->
-                    onShowConnectEvent.postValue(ViewModelEvent(connectionAppLinkData))
+                    onShowConnectEvent.postValue(ViewModelEvent(Bundle().apply {
+                        putSerializable(KEY_DATA, connectionAppLinkData)
+                    }))
                 } ?: intent.deepLink.extractActionAppLinkData()?.let { actionAppLinkData ->
-                    onShowSubmitActionEvent.postValue(ViewModelEvent(actionAppLinkData))
+                    onShowSubmitActionEvent.postValue(ViewModelEvent(Bundle().apply {
+                        putSerializable(KEY_ACTION_DEEP_LINK_DATA, actionAppLinkData)
```

Comment:
line:130 `connectionAppLinkData` is assigned to `KEY_DATA`  
line:133 `actionAppLinkData ` is assigned to `KEY_ACTION_DEEP_LINK_DATA`  
please unify keys names. 
I think `KEY_DATA`  is enough for both cases